lib/auto-languageclient.ts

```diff
@@ -47,33 +47,40 @@ import {
 export { ActiveServer, LanguageClientConnection, LanguageServerProcess };
 export type ConnectionType = 'stdio' | 'socket' | 'ipc';
 
+export interface ServerAdapters {
+  linterPushV2: LinterPushV2Adapter;
+  loggingConsole: LoggingConsoleAdapter;
+  signatureHelpAdapter?: SignatureHelpAdapter;
+}
+
 // Public: AutoLanguageClient provides a simple way to have all the supported
 // Atom-IDE services wired up entirely for you by just subclassing it and
 // implementing startServerProcess/getGrammarScopes/getLanguageName and
 // getServerName.
 export default class AutoLanguageClient {
-  private _disposable = new CompositeDisposable();
-  private _serverManager: ServerManager;
-  private _consoleDelegate: atomIde.ConsoleService;
-  private _linterDelegate: linter.IndieDelegate;
-  private _signatureHelpRegistry: atomIde.SignatureHelpRegistry | null;
-  private _lastAutocompleteRequest: AutocompleteRequest;
-  private _isDeactivating: boolean;
+  private _disposable!: CompositeDisposable;
```

Comment:
Lest I forget, this is actually a bug. If `_disposable` is created on `AutoLanguageClient` construction, and dismissed in `deactivate()`, an activation after deactivation will error due to `add`ing to a disposed `CompositeDisposable`. Hence `_disposable` has to be created in `activate()`.